File: events/src/main/java/com/nerdscorner/mvplib/events/model/BaseEventsModel.java
package com.nerdscorner.mvplib.events.model;

import com.nerdscorner.mvplib.commons.mvp.model.BaseModel;
import com.nerdscorner.mvplib.events.bus.Bus;

public class BaseEventsModel extends BaseModel {

    private Bus bus;

    public BaseEventsModel() {
        bus = Bus.getDefaultEventBus();
    }

    public BaseEventsModel(Bus bus) {
        this.bus = bus;
    }

    public void setBus(Bus bus) {
        this.bus = bus;
    }

    public Bus getBus() {
        return bus;
    }
}

File: events/src/main/java/com/nerdscorner/mvplib/events/model/BaseEventsModel.kt
package com.nerdscorner.mvplib.events.model

import com.nerdscorner.mvplib.events.bus.Bus

open class BaseEventsModel(@JvmField protected var bus: Bus = Bus.defaultBus) {

    @JvmName("setBusValue")
    fun setBus(bus: Bus) {
        this.bus = bus
    }

    @JvmName("getBusValue")
    fun getBus() = bus

    fun removeStickyEvent(event: Any) {
        bus.removeStickyEvent(event)
    }
}
